File: CCCC天梯赛/c4天梯-练习/L2-017.cpp
//2018-03-14
#include <cstdio>
#include <cstring>
#include <iostream>
#include <algorithm>

using namespace std;

const int N = 100005;
int arr[N];

int main()
{
	int n;
	while(~scanf("%d", &n)){
		for(int i = 0; i < n; i++)
		  	scanf("%d", &arr[i]);
		sort(arr, arr+n);
		int in = n/2;
		int sumIn = 0, sumOut = 0;
		for(int i= 0; i < n; i++){
			if(i < in) sumIn += arr[i];
			else sumOut += arr[i];
		}
		printf("Outgoing #: %d\nIntroverted #: %d\nDiff = %d\n", n-in, in, sumOut-sumIn);
	}

	return 0;
}

File: CCCC天梯赛/c4天梯-练习/L2-017.java
import java.util.Arrays;
import java.util.Scanner;

public class Main
{
	static final int N = 100005;
    public static void main(String[] args)
    {
    	Scanner cin = new Scanner(System.in);
        int n;
        int[] arr = new int[N];
        while(cin.hasNext()) {
        	n = cin.nextInt();
        	for(int i = 0; i < n; i++)
        		arr[i] = cin.nextInt();
        	Arrays.sort(arr, 0, n);
        	int in = n/2;
        	int sumIn = 0, sumOut = 0;
        	for(int i = 0; i < n; i++)
        		if(i < in) sumIn += arr[i];
        		else sumOut += arr[i];
        	System.out.printf("Outgoing #: %d\nIntroverted #: %d\nDiff = %d\n", n-in, in, sumOut-sumIn);
        }
    }
}
